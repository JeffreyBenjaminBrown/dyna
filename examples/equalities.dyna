% Given two items with different names and equal values...
a += 0.
b += 0.

% ==
by_evl_refl  := a == a.     % True
by_evl_cross := a == b.     % True

% =
by_syn_refl  := a =  a.     % &a is the same as &a
by_syn_cross := a =  b.     % ... but not the same as &b.

% is
by_is_0a     := 0 is a.     % That's right, a evaluates to 0.
by_is_ab     := a is b.     % b does not evaluate to &a, so this is
                            % a unification failure and the head
                            % remains null (not false).
