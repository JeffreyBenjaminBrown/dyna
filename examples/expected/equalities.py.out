
Solution
========
a = 0.
b = 0.
by_evl_cross = true.
by_evl_refl = true.
by_is_0a = true.
by_syn_cross = false.
by_syn_refl = true.

